src/JBMultiTerminal.sol

```diff
@@ -555,31 +576,43 @@ contract JBMultiTerminal is JBPermissioned, ERC2771Context, IJBMultiTerminal {
         if (ruleset.id != 0 && !ruleset.allowAddAccountingContext()) revert ADDING_ACCOUNTING_CONTEXT_NOT_ALLOWED();
 
         // Keep a reference to the number of accounting contexts to add.
-        uint256 numberOfAccountingContexts = tokens.length;
+        uint256 numberOfAccountingContexts = accountingContexts.length;
 
-        // Keep a reference to the token being iterated on.
-        address token;
+        // Keep a reference to the accounting context being iterated on.
+        JBAccountingContext memory accountingContext;
 
         // Start accepting each token.
         for (uint256 i; i < numberOfAccountingContexts; i++) {
             // Set the accounting context being iterated on.
-            token = tokens[i];
+            accountingContext = accountingContexts[i];
 
             // Get a storage reference to the currency accounting context for the token.
-            JBAccountingContext storage accountingContext = _accountingContextForTokenOf[projectId][token];
+            JBAccountingContext storage storedAccountingContext =
+                _accountingContextForTokenOf[projectId][accountingContext.token];
 
             // Make sure the token accounting context isn't already set.
-            if (accountingContext.token != address(0)) revert ACCOUNTING_CONTEXT_ALREADY_SET();
+            if (storedAccountingContext.token != address(0)) revert ACCOUNTING_CONTEXT_ALREADY_SET();
+
+            // Make sure the decimals provided in the accounting context matches expectations if there are expectations.
+            if (
+                (accountingContext.token == JBConstants.NATIVE_TOKEN && accountingContext.decimals != 18)
+                    || (
+                        IERC165(accountingContext.token).supportsInterface(type(IERC20Metadata).interfaceId)
+                            && accountingContext.decimals != IERC20Metadata(accountingContext.token).decimals()
+                    )
+            ) {
+                revert INVALID_ACCOUNTING_CONTEXT_DECIMALS();
```

Comment:
LGTM except this. decimals can be manipulated by a custom impl (overridden) decimals() of an erc20. 

It's also kind of a gotcha from UX perspective that they could enter the wrong decimals when we could just grab that for them. 

So I say check for interface support -> grab decimals = reduce calldata and gotchas.

Reply:
... we first check to see if the token reports `decimals()`, if so we make sure the provided value aligns.

Reply:
<img width="745" alt="image" src="https://github.com/Bananapus/nana-core/assets/77952627/9804e884-29fc-40eb-91b8-5f1c36b85e38">

Reply:
the thing now is that we also allow the client to provide custom value in case the token doesn't report